### ISSUE TITLE
Bloated Repository

### ISSUE THREAD

**Issue by agrawal-d** [crux-bphc/fest-manager#350]
The commits for the previous fests have made the repository extremely bloated. Cloning the master branch takes up a **lot** of bandwidth. Also, unrequited resources like images are also present in previous commits.

I do not know the best practice to fix this. One way may be to reset the master branch to only the current HEAD and remove all commit histories, but this does not seem ideal.

Any possible fix for this?

**Comment by Nischay-Pro** [comment]
`git clone https://github.com/crux-bphc/fest-manager.git --depth=1` to fetch the latest commit only
`git clone https://github.com/crux-bphc/fest-manager.git --depth=100` to fetch the latest 100 commits.

**Comment by agrawal-d** [comment]
@Nischay-Pro Yes. I am aware of this. But this is not the issue I am concerned about. I think that the master branch should only contain the development code, and not the deployment resources, such as the static pages, post-holder images etc. These, according to me, should be in their own branches. This is being done now, but previously the work was done on the master branch.

I was thinking of removing those resources from the master branch to de-bloat it.

**Comment by aero31aero** [comment]
There's one way here that would be sort of ideal. Git allows you to rewrite all history, and usually we ask you to not rewrite master; however we can remove objects from past commits to act as if they were never a part of the repository.

Here's what I would do:

0. On master, rung `git branch -B backup` in case something goes wrong.
1. Make the /images folder be split into two: /images/core and /images/custom. In the core, place all resources that we want to keep in the repo: Icons for google login, maybe fontawesome etc if we have those, or some other basic resources. Rewrite the code for that. Move all things like logo, homepage images etc to `custom` and gitignore it.
2. Commit the above changes.
3. Run `git filter-branch --tree-filter 'rm -f <some path here>' -- --all` such that the rm command would remove all images except those in /core from all the commits.
4. Keep in mind that the above would rewrite all commits like a rebase does, so now you force push the master branch to github.
5. Carefully point all the tags for various other fests to their relevant commits.
6. Delete any branches other than master.

If you're successful, a fresh clone wouldn't have any image blobs and would be smaller. If you mess up, reset master from backup branch and retry.

**Comment by aero31aero** [comment]
> These, according to me, should be in their own branches. This is being done now, but previously the work was done on the master branch.

Aha! Nice! I'd be interested to see what approach you take with this. Best of luck!
